File: src/test/resources/IsStringUnique2.java
public class Solution {
    /**
     * @param str input string to be checked
     * @return  Indicate if string contains only unique chars
     */
    public boolean isUniqueChars(String str) {
        if (str == null) return true;
        if (str.length() > 128) return false;

        boolean[] charSet = new boolean[128];

        for (int i = 0; i < str.length(); i++) {
            int val = str.charAt(i);
            if (charSet[val]) return false;

            charSet[val] = true;
        }

        return true;
    }
}

File: src/test/resources/IsStringUnique2.kt
class IsStringUnique2 {
    /**
     * @param str input string to be checked
     * *
     * @return  Indicate if string contains only unique chars
     */
    fun isUniqueChars(str: String?): Boolean {
        if (str == null) return true
        if (str.length > 128) return false

        val charSet = BooleanArray(128)

        for (i in 0..str.length - 1) {
            val value = str[i].toInt()
            if (charSet[value]) return false

            charSet[value] = true
        }

        return true
    }
}